src/olympia/hero/tests/test_models.py

```diff
@@ -231,6 +232,29 @@ def test_clean_cta_remove_prefixes(self):
             hero.clean()
             assert hero.cta_url == '/en-US/firefox/something/weird/'
 
+    def test_clean_cta_remove_prefixes_with_site_url(self):
+        hero = SecondaryHero.objects.create()
+
+        with self.activate(locale='en-US', app='firefox'):
+            # With default test settings, this should be transformed because
+            # SITE_URL is http://testserver.
+            hero.cta_url = 'http://testserver/en-US/firefox/addon/foo'
+            hero.clean()
+            assert hero.cta_url == '/addon/foo/'
+
+        with self.activate(locale='en-US', app='firefox'):
+            # With default test settings, this isn't transformed because
+            # neither the SITE_URL nor the EXTERNAL_SITE_URL match.
+            hero.cta_url = 'https://addons.mozilla.org/en-US/firefox/addon/foo'
+            hero.clean()
+            assert hero.cta_url == 'https://addons.mozilla.org/en-US/firefox/addon/foo'
+
+            # When overriding the EXTERNAL_SITE_URL however, it works.
+            with override_settings(EXTERNAL_SITE_URL='https://addons.mozilla.org'):
+                hero.cta_url = 'https://addons.mozilla.org/en-US/firefox/addon/foo'
```

Comment:
nit: `hero.cta_url` was already that value - it was `assert`ed that it was on line 250.

Reply:
I know, but I figured it was a bit more explicit and easier to cut&paste that way.  (I almost made a bunch of different methods - I would have loved to use `pytest.mark.parametrize` but it doesn't play nice with methods vs functions)